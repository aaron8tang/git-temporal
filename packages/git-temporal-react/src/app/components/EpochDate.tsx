import React from 'react';
import { style } from 'app/styles';

export interface EpochDateProps {
  epochTime: number;
  style?: object;
}

const dateDisplayOptions = { year: 'numeric', month: 'long', day: 'numeric' };

// numbers for humans have commas
export const EpochDate = (props: EpochDateProps): JSX.Element => {
  const humanDate = new Date(props.epochTime * 1000).toLocaleDateString(
    'en-US',
    dateDisplayOptions
  );
  return <span style={style(props.style)}>{humanDate}</span>;
};
