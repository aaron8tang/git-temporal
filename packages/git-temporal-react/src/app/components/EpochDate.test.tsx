import React from 'react';
import { shallow } from 'enzyme';

import { EpochDate } from './EpochDate';

describe('components/EpochDate', () => {
  describe('when rendered epoch time 1539369263 (Fri, 12 Oct 2018 18:34:23 GMT)', () => {
    let wrapper;
    beforeAll(() => {
      wrapper = shallow(<EpochDate epochTime={1539369263} />);
    });

    test('it should match snapshot (it should have rendered "3 weeks")', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
