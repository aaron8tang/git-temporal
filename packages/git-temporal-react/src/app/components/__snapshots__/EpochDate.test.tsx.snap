// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/EpochDate when rendered epoch time 1539369263 (Fri, 12 Oct 2018 18:34:23 GMT) it should match snapshot (it should have rendered "3 weeks") 1`] = `
<span
  style={Object {}}
>
  October 12, 2018
</span>
`;
